kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ResourcesUtil.java

```diff
@@ -277,9 +277,7 @@ static <O extends HasMetadata, R extends HasMetadata> Set<ResourceID> findReferr
                 primary -> refAccessor.apply(primary).stream().anyMatch(ref -> ResourcesUtil.isReferent(ref, referent)));
     }
 
-    static List<Condition> maybeAddOrUpdateCondition(List<Condition> conditions, Condition condition) {
-        var type = Objects.requireNonNull(condition.getType());
-
+    static List<Condition> maybeAddOrUpdateCondition(List<Condition> conditions, List<Condition> conditionsToAdd) {
```

Comment:
This is ripe for a smaller scoped PR to tidy it up. There's a lot of pain being inflicted by treating the conditions as a `List<Condition>`. The semantics we really want are a `Map<Condition.Type, Condition>`, with APIs to match. But I think this is a follow on PR.